internal/restorer/restorer.go

```diff
@@ -305,6 +307,59 @@ func (res *Restorer) RestoreTo(ctx context.Context, dst string) error {
 	return err
 }
 
+// verifyFile checks if the given target file's contents match the
+// blobs given in node
+// It first checks the file size and then checks all file blobs
+// It always returns the successfully checked blobs
+func (res *Restorer) verifyFile(target string, node *restic.Node) (matchingBlobs []bool, err error) {
+	debug.Log("calling verifyFile")
+
+	matchingBlobs = make([]bool, len(node.Content))
+	stat, err := os.Stat(target)
```

Comment:
Shouldn't this be Lstat? Otherwise, a symlink to a matching file is considered a match.

Reply:
I actually have a use case for that: There some exceptionally large files that I already have restored and redeployed elsewhere in the filesystem. With this I could symlink it within the restore directory and not worry about having two copies or having it restored again. Perhaps a command line flag to ignore symlinks would be helpful for the opposite case?

Reply:
You are right. I just copied this from the already existing `VerifyFiles` but in fact, this won't correctly verify restored files if they were exchanged by symlinks (whoever would 

 Changing this here also is not sufficient as the file reading part below would also follow symlinks. I just realized that the archiver code uses `arch.FS.OpenFile(target, fs.O_RDONLY|fs.O_NOFOLLOW, 0)`, so I think this should be changed too, here.

@ianmaddox I would not implement your use case in this PR. IMO it is an additional feature. So far, `restore` always will *exactly* reproduce the snapshot contents.

Maybe you open a separate issue for your use case after this PR is finished and discuss this independent from here?